src/app/page.tsx

```diff
@@ -1,91 +1,69 @@
 "use client";
 
 import { useEffect, useState } from "react";
+import { AdvocatesTable } from "@/components/AdvocatesTable";
+import { SearchInput } from "@/components/SearchInput";
+import { AdvocateResultsT } from "@/app/api/advocates/route";
+import { useToast } from "@/hooks/use-toast";
+import { useDebounce } from "use-debounce";
+import { Loader2 } from "lucide-react";
 
 export default function Home() {
-  const [advocates, setAdvocates] = useState([]);
-  const [filteredAdvocates, setFilteredAdvocates] = useState([]);
+  const [advocateData, setAdvocateData] = useState<
+    AdvocateResultsT["data"] | undefined
+  >();
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const [currentPage, setCurrentPage] = useState(1);
+  const [debouncedSearchTerm] = useDebounce(searchTerm, 500);
```

Comment:
Added some debouncing here to avoid making requests on each keystroke. 500ms generally feels good to me, but can be tweaked if needed.